test-jest/chart/LineChart.spec.tsx

```diff
@@ -522,100 +412,109 @@ describe('<LineChart /> - Rendering two line charts with syncId', () => {
     { name: 'Page A', uv: 230, pv: 2400, amt: 2400 },
   ];
 
-  const runAllPromises = () => new Promise(resolve => setTimeout(resolve));
+  // const runAllPromises = () => new Promise(resolve => setTimeout(resolve));
 
-  test.skip('should show tooltips for both charts synced by index on MouseEnter and hide on MouseLeave/Escape', async () => {
-    const ActiveDot = ({ cx, cy }) => <circle cx={cx} cy={cy} r={10} className="customized-active-dot" />;
-
-    const chart1 = (
-      <LineChart width={width} height={height} data={data} margin={margin} syncId="test">
-        <Line activeDot={<ActiveDot />} type="monotone" dataKey="uv" stroke="#ff7300" />
-        <Tooltip />
-        <XAxis dataKey="name" />
-      </LineChart>
-    );
-
-    const chart2 = (
-      <LineChart width={width} height={height} data={data2} margin={margin} syncId="test">
-        <Line activeDot={<ActiveDot />} type="monotone" dataKey="uv" stroke="#ff7300" />
-        <Tooltip />
-        <XAxis dataKey="name" />
-      </LineChart>
-    );
+  // type ActiveDotProps = {
+  //   cx?: number;
+  //   cy?: number;
+  // };
 
+  // eslint-disable-next-line max-len
+  test.skip('should show tooltips for both charts synced by index on MouseEnter and hide on MouseLeave/Escape', async () => {
```

Comment:
I am having trouble triggering the tool-tip for those tests. Tried different approaches, `fireEvent` with `pageX` & `pageY` manually set, [RTL User Interaction Hover](https://testing-library.com/docs/user-event/convenience#hover) but I cannot make the '.recharts-tooltip-cursor' pop. Can anybody give me a hand?

Reply:
We might hit some limitations of `jsdom` here - try logging and seeing if you're getting any actual values set for pageX and pageY

Usually for `fireEvent` I will do something like the following:

```ts
const firstChart = container.querySelector('.recharts-wrapper');

fireEvent.mouseEnter(firstChart);
```

Reply:
In this case, pageX will be undefined. Was the same for the Brush test, where I had to set pageX using Object.assign as you cannot set it directly with `fireEvent`. where it works `mousedown` and `mousemove` I cannot make it work for `mouseEnter`. 

https://github.com/recharts/recharts/pull/3141/files#diff-467d743ad34119402d9ef9d760c90b7ba4053cab936f4fa4288db04b7fecef2eR364

I will do some logs later and see what happens.

Reply:
I also looked into this and had the same issue. 
You could consider to solve the issue in two PRs, and solve the Tooltip tests in another PR, still getting in most changes now.

Reply:
Hmm, this might be why karma was used initially. Karma uses Headless or actual Chrome/Firefox to run tests - Jest uses `jsdom` which doesn't use the actual browser, it just acts like it.

Adding E2E type tests with something like Cypress/Puppeteer/Selenium/etc. could test for these things but that would be a large and separate effort + more things to maintain

Reply:
I have good news, I can make them pass. The following console.log is set here https://github.com/recharts/recharts/blob/master/src/chart/generateCategoricalChart.tsx#L1349 inside the handler

`fireEvent.mouseEnter(firstChart);` logs -> `pageX:  undefined`

```javascript
const mouseEnterEvent = new MouseEvent('mouseenter', { bubbles: true, cancelable: true });
    Object.assign(mouseEnterEvent, { pageX: 100, pageY: height / 2 });
    fireEvent(firstChart, mouseEnterEvent);
```
doesn't log, the event is not triggered, however,

```javascript
const mouseEnterEvent = new MouseEvent('mouseover', { bubbles: true, cancelable: true });
    Object.assign(mouseEnterEvent, { pageX: 100, pageY: height / 2 });
    fireEvent(firstChart, mouseEnterEvent);
``` 

logs -> `pageX:  100`

So I can use `mouseover` instead of `mouseenter`.

There is another complexity, I have to mock the `window.requestAnimationFrame` function, otherwize it will not pop the second popup when using `syncId`. I will finish this PR over the weekend.